src/lib/layouts/workflow-history-layout-v2.svelte

```diff
@@ -0,0 +1,203 @@
+  import ToggleButtons from '$lib/holocene/toggle-button/toggle-buttons.svelte';
+  import ToggleSwitch from '$lib/holocene/toggle-switch.svelte';
+  import { translate } from '$lib/i18n/translate';
+  import { groupEvents } from '$lib/models/event-groups';
+  import {
+    activeEvents,
+    activeGroups,
+    clearActives,
+  } from '$lib/stores/active-events';
+  import { eventViewType } from '$lib/stores/event-view';
+  import {
+    decodeEventHistory,
+    filteredEventHistory,
+    fullEventHistory,
+  } from '$lib/stores/events';
+  import { workflowRun } from '$lib/stores/workflow-run';
+  import { decodeURIForSvelte } from '$lib/utilities/encode-uri';
+  import { exportHistory } from '$lib/utilities/export-history';
+  import { getWorkflowTaskFailedEvent } from '$lib/utilities/get-workflow-task-failed-event';
+
+  let showFilters = false;
+  let zoomLevel = 1;
+
+  $: ({ namespace } = $page.params);
+  $: ({ workflow } = $workflowRun);
+  $: pendingActivities = workflow?.pendingActivities;
+  $: groups = groupEvents(
+    $filteredEventHistory,
+    'ascending',
+    pendingActivities,
+  );
+
+  $: workflowTaskFailedError = getWorkflowTaskFailedEvent(
+    $fullEventHistory,
+    'ascending',
+  );
+
+  $: $eventViewType, clearActives();
+
+  beforeNavigate(() => {
+    clearActives();
+  });
+
+  let canvasWidth = 0;
+  let showDownloadPrompt = false;
+
+  const onDownloadClick = () => {
+    showDownloadPrompt = false;
+    exportHistory({
+      namespace: decodeURIForSvelte(namespace),
+      workflowId: decodeURIForSvelte(workflow?.id),
+      runId: decodeURIForSvelte(workflow?.runId),
+      settings: $page.data.settings,
+      decodeEventHistory: $decodeEventHistory,
+    });
+  };
+
+  const zoomOut = () => {
+    if (zoomLevel < 10) zoomLevel += 0.5;
+  };
+
+  const zoomIn = () => {
+    if (zoomLevel > 1) {
+      zoomLevel -= 0.5;
+    }
+  };
+</script>
+
+<div class="surface-secondary flex flex-col gap-0 px-8">
+  <WorkflowCallStackError />
+  <div class="flex flex-col gap-2">
+    <WorkflowDetails />
+    <InputAndResults />
+    {#if workflowTaskFailedError}
+      <WorkflowError error={workflowTaskFailedError} />
+    {/if}
+    <div
+      class="flex flex-col items-center justify-between gap-2 py-2 md:flex-row"
+    >
+      <div class="flex flex-col items-center gap-2 md:flex-row md:gap-4">
+        <h2 class="text-2xl font-medium">
+          {translate('workflows.event-history')}
+        </h2>
+        <ToggleButtons>
+          <ToggleButton
+            active={$eventViewType === 'compact'}
+            data-testid="compact"
+            on:click={() => ($eventViewType = 'compact')}
+            >{translate('workflows.compact')}</ToggleButton
+          >
+          <ToggleButton
+            active={$eventViewType === 'timeline'}
+            data-testid="timeline"
+            on:click={() => ($eventViewType = 'timeline')}
+            >{translate('common.timeline')}</ToggleButton
+          >
```

Comment:
NIT: Noticed that if `Timeline` is selected and then labs mode is turned off `History` does not show up as selected (although it does default to the History view).

<img width="598" alt="Screenshot 2024-04-04 at 5 18 49 PM" src="https://github.com/temporalio/ui/assets/15069288/8f32730e-fb10-4594-8da0-ad31f05533d0">

Reply:
👍 Fixed to show History as active when Timeline is the view but not on labs mode